### ISSUE TITLE
Thank you!

### ISSUE THREAD

**Issue by catid** [snu-mllab/Achievement-Distillation#1]
Just wanted to say thank you for sharing this project, which I'm using as a starting point to learning about RL.

Since I spent most of last year playing with LLMs, I'd like to figure out how to put together a project with a self-hosted Mixtral LLM that uses your RL software to try to solve Microsoft TextWorld challenges.  My goal is to treat this as a toy example for learning the basics.  Do you have any pointers to how best to do this?

**Comment by symoon11** [comment]
Thank you for your interest in this project. While I’m not an expert in LLMs, I’m glad to provide some useful pointers to help you get started.

1. I recommend starting with the following repositories. These will help you understand the process of fine-tuning LLMs using PPO. Here, FLAN-T5 is used as the base model.
   - https://github.com/flowersteam/lamorel
   - https://github.com/flowersteam/Grounding_LLMs_with_online_RL
2. Once you have a good grasp on this, you'll be ready to experiment with Microsoft TextWorld. Switching to Mixtral as the base model is straightforward. To do this, simply update the model path in the following configuration file.
   - https://github.com/flowersteam/lamorel/blob/main/examples/PPO_LoRA_finetuning/local_gpu_config.yaml
3. If you succeed in obtaining a LLM policy that performs reasonably well on Microsoft TextWorld, then you can apply my method to enhance it further.

Feel free to ask if you have more questions and I would greatly appreciate it if you could share any interesting results.

**Comment by catid** [comment]
Thanks for sharing these papers!

The direction I ended up going in is an older idea that's been tried a few times.  The latest implementation I could find was this one: https://github.com/princeton-nlp/calm-textgame

So the toy project has become how to adapt this to modern primitives like `sentence-transformers`, Mixtral, LRU instead of GRU, AD-PPO in place of GRRN.

I'd like to avoid retraining the LLM itself so that it can be an agentic system that can load different RL modules based on familiar tasks, and hopefully to reduce the costs of training.  Also modifying the LLM is a bit tricky since it messes with the exllamav2 code that's harder to hack on for a novice like me.

**Comment by symoon11** [comment]
Sorry for the late reply.

I think this would be a good starting point. It seems DRRN is an algorithm based on Q-learning, whereas PPO and AD-PPO are policy gradient algorithms. Transitioning from Q-learning to policy gradient could potentially lead to a degrade in performance. Nonetheless, AD can also be integrated into Q-learning based algorithms (please refer to Section 5.6 of the paper). I hope this helps you out!
